packages/gui/src/components/nfts/gallery/NFTGallery.tsx

```diff
@@ -148,6 +153,44 @@ export default function NFTGallery() {
 
   const selectedAll = useMemo(() => selectedVisibleNFTs.length === nfts.length, [nfts, selectedVisibleNFTs]);
 
+  const nftsSorted = React.useRef<any>(null);
+
+  const lastSelectedUserFolder = React.useRef<string | null>();
+
+  const isSortableActive = React.useRef<boolean>(false);
+
+  const { data: fingerprint } = useGetLoggedInFingerprintQuery();
+
+  React.useEffect(() => {
+    if (
+      lastSelectedUserFolder.current &&
+      userFolder !== lastSelectedUserFolder.current &&
+      nftsSorted.current?.destroy &&
+      isSortableActive.current
+    ) {
+      isSortableActive.current = false;
+      nftsSorted.current?.destroy();
+    }
+    lastSelectedUserFolder.current = userFolder;
+    const sortableParent = document.querySelector('[data-test-id=virtuoso-item-list]');
```

Comment:
do not use document.anything directly
use react standard useRef, ref for this everywhere